src/ralph/admin/filters.py

```diff
@@ -0,0 +1,172 @@
+# -*- coding: utf-8 -*-
+import re
+from datetime import datetime
+from functools import lru_cache
+
+from django.contrib.admin import SimpleListFilter
+from django.db.models import Q
+from django.utils.encoding import smart_text
+from django.utils.formats import get_format
+from django.utils.translation import ugettext_lazy as _
+
+
+@lru_cache()
+def date_format_to_human(value):
+    """
+    Convert Python format date to human.
+
+    Example:
+        >>> date_format_to_human('%Y-%m-%d')
+        YYYY-MM-DD
+
+    :param value: Date format example: %Y-%m
+    """
+    maps = {
+        '%y': 'YY',
+        '%Y': 'YYYY',
+        '%m': 'MM',
+        '%d': 'DD',
+    }
+    for k, v in maps.items():
+        value = value.replace(k, v)
+    return value
+
+
+class BaseCustomFilter(SimpleListFilter):
+
+    def has_output(self):
+        return True
+
+    def lookups(self, request, model_admin):
+        return None
+
+
+class ChoicesFilter(BaseCustomFilter):
+
+    """Renders filter form with choices field."""
+
+    template = 'admin/filters/choices_filter.html'
+    choices_list = []
+
+    def choices(self, cl):
+        yield {
+            'selected': False,
+            'value': '',
+            'display': _('All'),
+            'parameter_name': self.parameter_name
+        }
+        for lookup, title in self.choices_list:
+            yield {
+                'selected': smart_text(lookup) == self.value(),
+                'value': lookup,
+                'display': title,
+            }
+
+    def queryset(self, request, queryset):
+        if self.value():
+            return queryset.filter(**{self.parameter_name: self.value()})
+
+
+class BooleanFilter(ChoicesFilter):
+
+    """Renders filter form with boolean field."""
+
+    choices_list = [
+        ('', _('All')),
+        ('1', _('Yes')),
+        ('0', _('No')),
+    ]
+
+    def choices(self, cl):
```

Comment:
is needed because BooleanFilter and ChoiceFilter returned are completely different values.